tracer/test/Datadog.Trace.Tests/Telemetry/TelemetrySettingsTests.cs

```diff
@@ -134,9 +133,72 @@ public void SetsTelemetryEnabledBasedOnApiKeyAndEnabledSettings(string apiKey, s
                 { ConfigurationKeys.ApiKey, apiKey },
             });
 
-            var settings = new TelemetrySettings(source, _tracerSettings);
+            var settings = TelemetrySettings.FromSource(source);
+            if (enabled && !string.IsNullOrEmpty(apiKey))
+            {
+                settings.Agentless.Should().NotBeNull();
+            }
+            else
+            {
+                settings.Agentless.Should().BeNull();
+            }
 
             settings.TelemetryEnabled.Should().Be(enabled);
+            settings.ConfigurationError.Should().BeNullOrEmpty();
+        }
+
+        [Theory]
+        [InlineData(null, null)]
+        [InlineData(null, true)]
+        [InlineData(null, false)]
+        [InlineData("SOMEKEY", true)]
+        [InlineData("SOMEKEY", false)]
+        public void SetsAgentlessBasedOnApiKey(string apiKey, bool? agentless)
```

Comment:
Can you add an additional XUnit test or additional test cases here where `ConfigurationKeys.Telemetry.Enabled == true` as opposed to not being set? In this specific scenario, there would *not* be agentless but `settings.TelemetryEnabled.Should().Be(true)`

Reply:
Added in https://github.com/DataDog/dd-trace-dotnet/pull/2617/commits/268f3caa33852467feec3a28e3e528b0011dfa40